src/lib/utils/index.ts

```diff
@@ -78,6 +78,8 @@ export const tokenOrUndefined = (assetReference: AssetReference | string) =>
 export const isSome = <T>(option: T | null | undefined): option is T =>
   !isUndefined(option) && !isNull(option)
 
+export const isTruthy = <T>(value: T | false): value is T => Boolean(value)
```

Comment:
this doesnt correctly check `value is T`, e.g if I do this 

```
const foo = isTruthy<string>(1234)
```

the compiler will think foo is string.

This is the closest you can get, but it excludes `NaN` which has no type for.
```
type Falsey = false | null | undefined | '' | 0
export const isTruthy = <T>(value: T | Falsey): value is T => Boolean(value)
```

Alternatively we can just ignore the generic since its mostly just used for filtering which takes boolean anyway:
```suggestion
export const isTruthy = (value: any): boolean => Boolean(value)
```

Reply:
I don't think this is true.

The compiler won't let you do `const foo = isTruthy<string>(1234)`, as it will error with `Argument of type '1234' is not assignable to parameter of type 'string | false'.`

The generic is required as `isTruthy`' is made to be passed into `filter<S extends T>(predicate: (value: T) => value is S): S[]` as the first arg, so it needs the type signature `(value: T) => boolean`.

Reply:
Ah got it, thank you for clarifying 🤝